python/ray/tests/test_advanced.py

```diff
@@ -442,15 +445,24 @@ def test_wait():
         run_test_in_multi_threads(test_wait, num_repeats=1)
 
     # Run tests in a driver.
-    test_api_in_multi_threads()
+    test_api_in_multi_threads_base()
+    if sys.platform != 'win32':
+        test_api_in_multi_threads_wait()
```

Comment:
I am not sure about just skipping this and the worker tests below. I think this is something different from multi-node, right?

In any case, it would be nice to preemptively error when this situation is occurring and not just hang (which the test does when not skipped). Where would be a good place to add a condition for this?

@wuisawesome ideas?

Reply:
Yeah, this test is not multi-node and should not be skipped. It would be worth trying to see why it is hanging if you can reproduce it on your machine.

Reply:
I can reproduce it. Even if I reduce the number of iterations to 4 and extend the timeouts, any use of wait fails.

Reply:
I opened #20173 for this, since it is a separate issue.